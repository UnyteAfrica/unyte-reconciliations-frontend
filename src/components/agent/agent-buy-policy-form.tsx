import { useEffect, useRef, useState } from "react";
import { twMerge } from "tailwind-merge";
import { CustomInput } from "../shared/input";
import { Icon } from "../shared/icon";
import { MdOutlineFileUpload } from "react-icons/md";
import { formatToNaira } from "@/utils/utils";
import { SlidingButton } from "../shared/sliding-button";
import toast from "react-hot-toast";
import { useQuery } from "@tanstack/react-query";
import { AgentQueryKeys } from "@/utils/query-keys";
import { AgentApi } from "@/services/api/api-agent";
import { QuoteFormValue } from "@/types/types";
import { Loader } from "../loader";
import { produce } from "immer";

export const BuyPolicyForm: React.FC<{
  policyType: string;
  isOpen: boolean;
  policyCategory: string;
  onClose: () => void;
}> = ({ policyType, isOpen, onClose, policyCategory }) => {
  const containerRef = useRef<HTMLDivElement>(null);
  const [formPage, setFormPage] = useState(1);
  const [isSubmitted, setIsSubmitted] = useState(false);
  const [customerFormValues, setCustomerFormValues] = useState<
    QuoteFormValue[]
  >([]);
  const [insuranceFormValues, setInsuranceFormValues] = useState<
    QuoteFormValue[]
  >([]);

  const policyCategoryId = policyCategory.split(" ")[0].toLowerCase();

  const {
    isPending: isLoadingQuoteParams,
    data: quoteParamsData,
    // error: quoteParamsError,
  } = useQuery({
    queryKey: [AgentQueryKeys.quoteParams],
    queryFn: () => AgentApi.getQuoteParams(policyCategoryId),
  });

  useEffect(() => {
    setCustomerFormValues(quoteParamsData?.customer_metadata || []);
  }, [quoteParamsData?.customer_metadata]);

  useEffect(() => {
    setInsuranceFormValues(quoteParamsData?.insurance_details || []);
  }, [quoteParamsData?.insurance_details]);

  return (
    <div
      className={twMerge(
        "z-50 h-dvh w-screen fixed top-0 left-0 flex flex-col bg-white py-10 px-6 pb-6 transition translate-x-[100%] opacity-0 overflow-y-auto",
        isOpen && "translate-x-0 opacity-100"
      )}
      ref={containerRef}
    >
      <div className="flex justify-between items-center mb-4">
        <button
          onClick={(e) => {
            e.stopPropagation();
            if (isSubmitted) {
              setIsSubmitted(false);
              return;
            }
            if (formPage == 1) {
              onClose();
            } else {
              setFormPage(1);
            }
          }}
        >
          <Icon type="arrowBack" className="" />
        </button>
        {!isSubmitted && (
          <>
            <div
              className={twMerge(
                "bg-[#D9D9D9] rounded-full h-[6px] w-[40%] transition",
                formPage == 2 && "bg-[#25D366]"
              )}
            />
            <div className="bg-[#D9D9D9] rounded-full h-[6px] w-[40%]" />
          </>
        )}
      </div>
      {isSubmitted && <PolicyPayment />}
      {!isSubmitted && (
        <>
          <header className="mb-4">
            <h4 className="font-semibold mb-2">Sell {policyType}</h4>
            <em className="not-italic text-sm text-[#4F4F4F]">
              Enter customer's personal details
            </em>
          </header>
          <div>
            {isLoadingQuoteParams ? (
              <Loader className="mx-auto w-16 h-16 mt-20" />
            ) : formPage == 1 ? (
              <div className="space-y-6">
                {customerFormValues.map((formValue, idx) => (
                  <CustomInput
                    key={idx}
                    label={formValue.name}
                    labelClassName="text-sm"
                    className="h-14 border-[#e0e0e0] text-sm"
                    value={formValue.value}
                    onChange={(e) =>
                      setCustomerFormValues(
                        produce((draft) => {
                          draft[idx].value = e.target.value;
                        })
                      )
                    }
                  />
                ))}
              </div>
            ) : (
              formPage == 2 && (
                <div className="space-y-6">
                  {insuranceFormValues.map((formValue, idx) => (
                    <CustomInput
                      key={idx}
                      label={formValue.name}
                      labelClassName="text-sm"
                      className="h-14 border-[#e0e0e0] text-sm"
                    />
                  ))}

                  <button className="w-full rounded-lg bg-[#F2F2F2] text-[#333] flex items-center justify-center py-4">
                    <MdOutlineFileUpload className="text-4xl inline-block mr-2" />
                    <span className="font-medium text-sm">
                      Upload Passport Data Page
                    </span>
                  </button>
                  <div className="flex justify-center items-center mt-4">
                    <Icon type="verified" className="inline-block mr-1" />
                    <em className="not-italic text-[#333] text-sm">
                      Your data is safe and secure
                    </em>
                  </div>
                </div>
              )
            )}
          </div>
          <div className="grow min-h-10" />
          <button
            onClick={(e) => {
              e.preventDefault();
              if (formPage == 1) {
                setFormPage(2);
                containerRef.current?.scrollTo({
                  top: 0,
                  left: 0,
                  behavior: "smooth",
                });
              } else {
                //Submit Form
                setIsSubmitted(true);
              }
            }}
            className="w-full bg-[#25D366] py-4 text-lg font-medium rounded-lg text-white"
          >
            {formPage == 1 ? "Continue" : "Get Quote"}
          </button>
          <em className="block text-center not-italic text-xs mt-10">
            Powered by Unyte Africa
          </em>
        </>
      )}
    </div>
  );
};

const PolicyPayment = () => {
  return (
    <div className="py-6 bg-white">
      <div className="w-full bg-[#C8FFC6] flex justify-center items-center mb-3 py-6 rounded-lg">
        <Icon type="travel" className="w-[100px] h-[100px]" />
      </div>
      <div className="py-[10px] mx-auto w-[165px] bg-[#101323] flex justify-center items-center rounded-full mb-3">
        <Icon type="whiteReceipt" className="mr-1 inline-block" />
        <span className="text-xs font-medium text-white">
          View coverage details
        </span>
      </div>
      <div className="bg-[#F9FAFB] rounded-lg p-4 space-y-4 mb-3">
        <div className="flex justify-between items-center">
          <em className="not-italic text-sm text-[#667085]">Policy number</em>
          <em className="not-italic text-sm font-medium text-[#101323]">
            **** **** 5678
          </em>
        </div>
        <div className="flex justify-between items-center">
          <em className="not-italic text-sm text-[#667085]">Effective date</em>
          <em className="not-italic text-sm font-medium text-[#101323]">
            19/09/2024
          </em>
        </div>
        <div className="flex justify-between items-center">
          <em className="not-italic text-sm text-[#667085]">Expiration Date</em>
          <em className="not-italic text-sm font-medium text-[#101323]">
            11/12/2024
          </em>
        </div>
        <div className="flex justify-between items-center">
          <em className="not-italic text-sm text-[#667085]">Cost</em>
          <em className="not-italic text-sm font-medium text-[#101323]">
            {formatToNaira(7500)}
          </em>
        </div>
      </div>
      <div className="space-y-[10px] bg-[#F9FAFB] rounded-lg p-4 mb-4">
        <em className="not-italic font-medium block">
          AXA Mansard Insurance LTD
        </em>
        <em className="not-italic font-medium block">3112456789</em>
        <em className="not-italic font-medium block">GTBank</em>
      </div>
      <em className="block mb-4 uppercase text-center not-italic">or</em>
      <Icon type="qr" className="mb-8 mx-auto" />
      <SlidingButton onSlide={() => toast.success("Policy Sold")} />
    </div>
  );
};
