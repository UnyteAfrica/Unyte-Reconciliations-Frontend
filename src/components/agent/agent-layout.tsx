import { Outlet } from "react-router-dom";
import { AgentNavbar } from "./agent-navbar";

export const AgentLayout = () => {
  return (
    <div>
      <AgentNavbar />
      <Outlet />
    </div>
  );
};
