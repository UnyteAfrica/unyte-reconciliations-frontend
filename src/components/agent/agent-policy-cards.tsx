import { Product } from "@/types/types";
import { PolicyCard } from "./agent-policy-card";
import { FaAngleLeft } from "react-icons/fa";

export const AgentPolicyCards: React.FC<{
  products: Product[];
  onBack: () => void;
}> = ({ products, onBack }) => {
  return (
    <main className="space-y-4">
      {products.map((product, idx) => (
        <PolicyCard policyType={product.name} key={idx} />
      ))}
      <button
        className="w-full text-white text-sm font-medium bg-[#25D366] rounded-xl p-4 flex justify-center items-center"
        onClick={onBack}
      >
        <FaAngleLeft />
        Back
      </button>
    </main>
  );
};
