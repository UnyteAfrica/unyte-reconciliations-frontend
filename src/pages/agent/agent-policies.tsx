import { AgentPoliciesTable } from "../../components/tables/policy-tables";
import { AgentPolicy } from "@/types/types";
import { PageContent } from "@/components/shared/page-content";

const policies: AgentPolicy[] = [
  {
    policyRef: "#WP62F3E8F93",
    policyNo: "123jkf5402",
    product: "Comprehensive",
    date: "May 7, 2023",
    price: 40000,
  },
  {
    policyRef: "#WP62F3E8F93",
    policyNo: "123jkf5402",
    product: "Credit Life",
    date: "May 7, 2023",
    price: 40000,
  },
  {
    policyRef: "#WP62F3E8F93",
    policyNo: "123jkf5402",
    product: "Card Protect",
    date: "May 7, 2023",
    price: 40000,
  },
  {
    policyRef: "#WP62F3E8F93",
    policyNo: "123jkf5402",
    product: "Student",
    date: "May 7, 2023",
    price: 40000,
  },
  {
    policyRef: "#WP62F3E8F93",
    policyNo: "123jkf5402",
    product: "Travel",
    date: "May 7, 2023",
    price: 40000,
  },
  {
    policyRef: "#WP62F3E8F93",
    policyNo: "123jkf5402",
    product: "Device",
    date: "May 7, 2023",
    price: 40000,
  },
];

export const AgentPolicies = () => {
  return (
    <PageContent
      title="Policies"
      pageTable={<AgentPoliciesTable policies={policies} />}
    />
  );
};
