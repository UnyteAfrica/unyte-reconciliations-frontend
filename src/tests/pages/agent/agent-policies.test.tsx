import { AgentPolicies } from "@/pages/agent/agent-policies";
import { render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";

describe("Agent Policies", () => {
  it("renders correctly", () => {
    const tree = render(
      <BrowserRouter>
        <AgentPolicies />
      </BrowserRouter>
    );
    expect(tree).toMatchSnapshot();
  });
});
