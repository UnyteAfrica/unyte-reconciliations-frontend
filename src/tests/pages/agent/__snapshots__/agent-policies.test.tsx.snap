// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Agent Policies > renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body
    style="overflow: unset;"
  >
    <div>
      <div
        class="px-5 pb-6 max-w-[850px] mx-auto bg-[#F8F8F8]"
      >
        <header
          class="mb-7"
        >
          <h1
            class="font-semibold text-2xl mb-5"
          >
            Policies
          </h1>
          <p>
            Choose any of our available policies to sell to potential customers
          </p>
        </header>
        <main
          class="space-y-4"
        >
          <div
            class="bg-white rounded-2xl p-4 cursor-pointer"
          >
            <div
              class="flex mb-2"
            >
              <div
                class="bg-[#C8FFC6] h-8 w-8 rounded-full mr-2 flex justify-center items-center shrink-0"
              >
                <img
                  alt=""
                  class=""
                  src="/src/assets/Icons/health-insurance.svg"
                />
              </div>
              <div>
                <h3
                  class="not-italic text-sm font-medium inline-block mb-1 text-[#101323]"
                >
                  Health Insurance
                </h3>
                <p
                  class="text-[#667085] text-xs"
                >
                  Covers medical expenses, such as doctor's visits, hospital stays, and prescription drugs.
                </p>
              </div>
            </div>
            <div
              class="flex justify-between"
            >
              <button
                class="w-[48%] text-white text-sm font-medium bg-[#25D366] rounded-full py-[6px]"
              >
                Sell Policy
              </button>
              <button
                class="w-[48%] text-black text-sm font-medium bg-[#F9FAFB] rounded-full py-[6px]"
              >
                View Policy
              </button>
            </div>
          </div>
          <div
            class="bg-white rounded-2xl p-4 cursor-pointer"
          >
            <div
              class="flex mb-2"
            >
              <div
                class="bg-[#C8FFC6] h-8 w-8 rounded-full mr-2 flex justify-center items-center shrink-0"
              >
                <img
                  alt=""
                  class=""
                  src="/src/assets/Icons/motor-insurance.svg"
                />
              </div>
              <div>
                <h3
                  class="not-italic text-sm font-medium inline-block mb-1 text-[#101323]"
                >
                  Comprehensive Motor Insurance
                </h3>
                <p
                  class="text-[#667085] text-xs"
                >
                  With motor insurance you get to cover damages caused by accidents or theft involving a vehicle.
                </p>
              </div>
            </div>
            <div
              class="flex justify-between"
            >
              <button
                class="w-[48%] text-white text-sm font-medium bg-[#25D366] rounded-full py-[6px]"
              >
                Sell Policy
              </button>
              <button
                class="w-[48%] text-black text-sm font-medium bg-[#F9FAFB] rounded-full py-[6px]"
              >
                View Policy
              </button>
            </div>
          </div>
          <div
            class="bg-white rounded-2xl p-4 cursor-pointer"
          >
            <div
              class="flex mb-2"
            >
              <div
                class="bg-[#C8FFC6] h-8 w-8 rounded-full mr-2 flex justify-center items-center shrink-0"
              >
                <img
                  alt=""
                  class=""
                  src="/src/assets/Icons/travel-insurance.svg"
                />
              </div>
              <div>
                <h3
                  class="not-italic text-sm font-medium inline-block mb-1 text-[#101323]"
                >
                  Travel Insurance
                </h3>
                <p
                  class="text-[#667085] text-xs"
                >
                  Travel insurance protects you against unexpected events while traveling, like trip cancellations, lost luggage etc.
                </p>
              </div>
            </div>
            <div
              class="flex justify-between"
            >
              <button
                class="w-[48%] text-white text-sm font-medium bg-[#25D366] rounded-full py-[6px]"
              >
                Sell Policy
              </button>
              <button
                class="w-[48%] text-black text-sm font-medium bg-[#F9FAFB] rounded-full py-[6px]"
              >
                View Policy
              </button>
            </div>
          </div>
          <div
            class="bg-white rounded-2xl p-4 cursor-pointer"
          >
            <div
              class="flex mb-2"
            >
              <div
                class="bg-[#C8FFC6] h-8 w-8 rounded-full mr-2 flex justify-center items-center shrink-0"
              >
                <img
                  alt=""
                  class=""
                  src="/src/assets/Icons/device-insurance.svg"
                />
              </div>
              <div>
                <h3
                  class="not-italic text-sm font-medium inline-block mb-1 text-[#101323]"
                >
                  Device Insurance
                </h3>
                <p
                  class="text-[#667085] text-xs"
                >
                  Device insurance covers your device against unexpected events that lead to damage of parts of the phone
                </p>
              </div>
            </div>
            <div
              class="flex justify-between"
            >
              <button
                class="w-[48%] text-white text-sm font-medium bg-[#25D366] rounded-full py-[6px]"
              >
                Sell Policy
              </button>
              <button
                class="w-[48%] text-black text-sm font-medium bg-[#F9FAFB] rounded-full py-[6px]"
              >
                View Policy
              </button>
            </div>
          </div>
        </main>
      </div>
    </div>
    <div
      class="z-50 h-dvh w-screen fixed top-0 left-0 flex flex-col bg-white py-10 px-6 pb-6 transition translate-x-[100%] opacity-0 overflow-y-auto"
    >
      <div
        class="flex justify-between items-center mb-4"
      >
        <button>
          <img
            alt=""
            class=""
            src="/src/assets/Icons/arrow-back.svg"
          />
        </button>
        <div
          class="bg-[#D9D9D9] rounded-full h-[6px] w-[40%] transition"
        />
        <div
          class="bg-[#D9D9D9] rounded-full h-[6px] w-[40%]"
        />
      </div>
      <header
        class="mb-4"
      >
        <h4
          class="font-semibold mb-2"
        >
          Sell 
          Health Insurance
        </h4>
        <em
          class="not-italic text-sm text-[#4F4F4F]"
        >
          Enter customer's personal details
        </em>
      </header>
      <div>
        <div
          class="space-y-6"
        >
          <div
            class=""
          >
            <label
              class="text-[#333] mb-2 block text-sm"
              for="customer'sFullName"
            >
              Customer's Full Name
               
            </label>
            <input
              class="block font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-14 border-[#e0e0e0] text-sm"
              id="customer'sFullName"
              placeholder="John Doe"
            />
            <p
              class="text-sm text-[#333]"
            />
          </div>
          <div
            class=""
          >
            <label
              class="text-[#333] mb-2 block text-sm"
              for="customer'sDateOfBirth"
            >
              Customer's Date of Birth
               
            </label>
            <input
              class="block font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-14 border-[#e0e0e0] text-sm"
              id="customer'sDateOfBirth"
              placeholder="DD/MM/YYYY"
            />
            <p
              class="text-sm text-[#333]"
            />
          </div>
          <div
            class=""
          >
            <label
              class="text-[#333] mb-2 block text-sm"
              for="customer'sPhoneNumber"
            >
              Customer's Phone Number
               
            </label>
            <input
              class="block font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-14 border-[#e0e0e0] text-sm"
              id="customer'sPhoneNumber"
              placeholder="08035892001"
            />
            <p
              class="text-sm text-[#333]"
            />
          </div>
          <div
            class=""
          >
            <label
              class="text-[#333] mb-2 block text-sm"
              for="customer'sEmailAddress"
            >
              Customer's Email Address
               
            </label>
            <input
              class="block font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-14 border-[#e0e0e0] text-sm"
              id="customer'sEmailAddress"
              placeholder="mail@mail.com"
            />
            <p
              class="text-sm text-[#333]"
            />
          </div>
        </div>
      </div>
      <div
        class="grow min-h-10"
      />
      <button
        class="w-full bg-[#25D366] py-4 text-lg font-medium rounded-lg text-white"
      >
        Continue
      </button>
      <em
        class="block text-center not-italic text-xs mt-10"
      >
        Powered by Unyte Africa
      </em>
    </div>
    <div
      class="z-50 h-dvh w-screen fixed top-0 left-0 flex flex-col bg-white py-10 px-6 pb-6 transition translate-x-[100%] opacity-0 overflow-y-auto"
    >
      <div
        class="flex justify-between items-center mb-4"
      >
        <button>
          <img
            alt=""
            class=""
            src="/src/assets/Icons/arrow-back.svg"
          />
        </button>
        <div
          class="bg-[#D9D9D9] rounded-full h-[6px] w-[40%] transition"
        />
        <div
          class="bg-[#D9D9D9] rounded-full h-[6px] w-[40%]"
        />
      </div>
      <header
        class="mb-4"
      >
        <h4
          class="font-semibold mb-2"
        >
          Sell 
          Comprehensive Motor Insurance
        </h4>
        <em
          class="not-italic text-sm text-[#4F4F4F]"
        >
          Enter customer's personal details
        </em>
      </header>
      <div>
        <div
          class="space-y-6"
        >
          <div
            class=""
          >
            <label
              class="text-[#333] mb-2 block text-sm"
              for="customer'sFullName"
            >
              Customer's Full Name
               
            </label>
            <input
              class="block font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-14 border-[#e0e0e0] text-sm"
              id="customer'sFullName"
              placeholder="John Doe"
            />
            <p
              class="text-sm text-[#333]"
            />
          </div>
          <div
            class=""
          >
            <label
              class="text-[#333] mb-2 block text-sm"
              for="customer'sDateOfBirth"
            >
              Customer's Date of Birth
               
            </label>
            <input
              class="block font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-14 border-[#e0e0e0] text-sm"
              id="customer'sDateOfBirth"
              placeholder="DD/MM/YYYY"
            />
            <p
              class="text-sm text-[#333]"
            />
          </div>
          <div
            class=""
          >
            <label
              class="text-[#333] mb-2 block text-sm"
              for="customer'sPhoneNumber"
            >
              Customer's Phone Number
               
            </label>
            <input
              class="block font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-14 border-[#e0e0e0] text-sm"
              id="customer'sPhoneNumber"
              placeholder="08035892001"
            />
            <p
              class="text-sm text-[#333]"
            />
          </div>
          <div
            class=""
          >
            <label
              class="text-[#333] mb-2 block text-sm"
              for="customer'sEmailAddress"
            >
              Customer's Email Address
               
            </label>
            <input
              class="block font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-14 border-[#e0e0e0] text-sm"
              id="customer'sEmailAddress"
              placeholder="mail@mail.com"
            />
            <p
              class="text-sm text-[#333]"
            />
          </div>
        </div>
      </div>
      <div
        class="grow min-h-10"
      />
      <button
        class="w-full bg-[#25D366] py-4 text-lg font-medium rounded-lg text-white"
      >
        Continue
      </button>
      <em
        class="block text-center not-italic text-xs mt-10"
      >
        Powered by Unyte Africa
      </em>
    </div>
    <div
      class="z-50 h-dvh w-screen fixed top-0 left-0 flex flex-col bg-white py-10 px-6 pb-6 transition translate-x-[100%] opacity-0 overflow-y-auto"
    >
      <div
        class="flex justify-between items-center mb-4"
      >
        <button>
          <img
            alt=""
            class=""
            src="/src/assets/Icons/arrow-back.svg"
          />
        </button>
        <div
          class="bg-[#D9D9D9] rounded-full h-[6px] w-[40%] transition"
        />
        <div
          class="bg-[#D9D9D9] rounded-full h-[6px] w-[40%]"
        />
      </div>
      <header
        class="mb-4"
      >
        <h4
          class="font-semibold mb-2"
        >
          Sell 
          Travel Insurance
        </h4>
        <em
          class="not-italic text-sm text-[#4F4F4F]"
        >
          Enter customer's personal details
        </em>
      </header>
      <div>
        <div
          class="space-y-6"
        >
          <div
            class=""
          >
            <label
              class="text-[#333] mb-2 block text-sm"
              for="customer'sFullName"
            >
              Customer's Full Name
               
            </label>
            <input
              class="block font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-14 border-[#e0e0e0] text-sm"
              id="customer'sFullName"
              placeholder="John Doe"
            />
            <p
              class="text-sm text-[#333]"
            />
          </div>
          <div
            class=""
          >
            <label
              class="text-[#333] mb-2 block text-sm"
              for="customer'sDateOfBirth"
            >
              Customer's Date of Birth
               
            </label>
            <input
              class="block font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-14 border-[#e0e0e0] text-sm"
              id="customer'sDateOfBirth"
              placeholder="DD/MM/YYYY"
            />
            <p
              class="text-sm text-[#333]"
            />
          </div>
          <div
            class=""
          >
            <label
              class="text-[#333] mb-2 block text-sm"
              for="customer'sPhoneNumber"
            >
              Customer's Phone Number
               
            </label>
            <input
              class="block font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-14 border-[#e0e0e0] text-sm"
              id="customer'sPhoneNumber"
              placeholder="08035892001"
            />
            <p
              class="text-sm text-[#333]"
            />
          </div>
          <div
            class=""
          >
            <label
              class="text-[#333] mb-2 block text-sm"
              for="customer'sEmailAddress"
            >
              Customer's Email Address
               
            </label>
            <input
              class="block font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-14 border-[#e0e0e0] text-sm"
              id="customer'sEmailAddress"
              placeholder="mail@mail.com"
            />
            <p
              class="text-sm text-[#333]"
            />
          </div>
        </div>
      </div>
      <div
        class="grow min-h-10"
      />
      <button
        class="w-full bg-[#25D366] py-4 text-lg font-medium rounded-lg text-white"
      >
        Continue
      </button>
      <em
        class="block text-center not-italic text-xs mt-10"
      >
        Powered by Unyte Africa
      </em>
    </div>
    <div
      class="z-50 h-dvh w-screen fixed top-0 left-0 flex flex-col bg-white py-10 px-6 pb-6 transition translate-x-[100%] opacity-0 overflow-y-auto"
    >
      <div
        class="flex justify-between items-center mb-4"
      >
        <button>
          <img
            alt=""
            class=""
            src="/src/assets/Icons/arrow-back.svg"
          />
        </button>
        <div
          class="bg-[#D9D9D9] rounded-full h-[6px] w-[40%] transition"
        />
        <div
          class="bg-[#D9D9D9] rounded-full h-[6px] w-[40%]"
        />
      </div>
      <header
        class="mb-4"
      >
        <h4
          class="font-semibold mb-2"
        >
          Sell 
          Device Insurance
        </h4>
        <em
          class="not-italic text-sm text-[#4F4F4F]"
        >
          Enter customer's personal details
        </em>
      </header>
      <div>
        <div
          class="space-y-6"
        >
          <div
            class=""
          >
            <label
              class="text-[#333] mb-2 block text-sm"
              for="customer'sFullName"
            >
              Customer's Full Name
               
            </label>
            <input
              class="block font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-14 border-[#e0e0e0] text-sm"
              id="customer'sFullName"
              placeholder="John Doe"
            />
            <p
              class="text-sm text-[#333]"
            />
          </div>
          <div
            class=""
          >
            <label
              class="text-[#333] mb-2 block text-sm"
              for="customer'sDateOfBirth"
            >
              Customer's Date of Birth
               
            </label>
            <input
              class="block font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-14 border-[#e0e0e0] text-sm"
              id="customer'sDateOfBirth"
              placeholder="DD/MM/YYYY"
            />
            <p
              class="text-sm text-[#333]"
            />
          </div>
          <div
            class=""
          >
            <label
              class="text-[#333] mb-2 block text-sm"
              for="customer'sPhoneNumber"
            >
              Customer's Phone Number
               
            </label>
            <input
              class="block font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-14 border-[#e0e0e0] text-sm"
              id="customer'sPhoneNumber"
              placeholder="08035892001"
            />
            <p
              class="text-sm text-[#333]"
            />
          </div>
          <div
            class=""
          >
            <label
              class="text-[#333] mb-2 block text-sm"
              for="customer'sEmailAddress"
            >
              Customer's Email Address
               
            </label>
            <input
              class="block font-sora w-full px-3 py-[16px] border rounded-lg outline-none focus:border-black h-14 border-[#e0e0e0] text-sm"
              id="customer'sEmailAddress"
              placeholder="mail@mail.com"
            />
            <p
              class="text-sm text-[#333]"
            />
          </div>
        </div>
      </div>
      <div
        class="grow min-h-10"
      />
      <button
        class="w-full bg-[#25D366] py-4 text-lg font-medium rounded-lg text-white"
      >
        Continue
      </button>
      <em
        class="block text-center not-italic text-xs mt-10"
      >
        Powered by Unyte Africa
      </em>
    </div>
  </body>,
  "container": <div>
    <div
      class="px-5 pb-6 max-w-[850px] mx-auto bg-[#F8F8F8]"
    >
      <header
        class="mb-7"
      >
        <h1
          class="font-semibold text-2xl mb-5"
        >
          Policies
        </h1>
        <p>
          Choose any of our available policies to sell to potential customers
        </p>
      </header>
      <main
        class="space-y-4"
      >
        <div
          class="bg-white rounded-2xl p-4 cursor-pointer"
        >
          <div
            class="flex mb-2"
          >
            <div
              class="bg-[#C8FFC6] h-8 w-8 rounded-full mr-2 flex justify-center items-center shrink-0"
            >
              <img
                alt=""
                class=""
                src="/src/assets/Icons/health-insurance.svg"
              />
            </div>
            <div>
              <h3
                class="not-italic text-sm font-medium inline-block mb-1 text-[#101323]"
              >
                Health Insurance
              </h3>
              <p
                class="text-[#667085] text-xs"
              >
                Covers medical expenses, such as doctor's visits, hospital stays, and prescription drugs.
              </p>
            </div>
          </div>
          <div
            class="flex justify-between"
          >
            <button
              class="w-[48%] text-white text-sm font-medium bg-[#25D366] rounded-full py-[6px]"
            >
              Sell Policy
            </button>
            <button
              class="w-[48%] text-black text-sm font-medium bg-[#F9FAFB] rounded-full py-[6px]"
            >
              View Policy
            </button>
          </div>
        </div>
        <div
          class="bg-white rounded-2xl p-4 cursor-pointer"
        >
          <div
            class="flex mb-2"
          >
            <div
              class="bg-[#C8FFC6] h-8 w-8 rounded-full mr-2 flex justify-center items-center shrink-0"
            >
              <img
                alt=""
                class=""
                src="/src/assets/Icons/motor-insurance.svg"
              />
            </div>
            <div>
              <h3
                class="not-italic text-sm font-medium inline-block mb-1 text-[#101323]"
              >
                Comprehensive Motor Insurance
              </h3>
              <p
                class="text-[#667085] text-xs"
              >
                With motor insurance you get to cover damages caused by accidents or theft involving a vehicle.
              </p>
            </div>
          </div>
          <div
            class="flex justify-between"
          >
            <button
              class="w-[48%] text-white text-sm font-medium bg-[#25D366] rounded-full py-[6px]"
            >
              Sell Policy
            </button>
            <button
              class="w-[48%] text-black text-sm font-medium bg-[#F9FAFB] rounded-full py-[6px]"
            >
              View Policy
            </button>
          </div>
        </div>
        <div
          class="bg-white rounded-2xl p-4 cursor-pointer"
        >
          <div
            class="flex mb-2"
          >
            <div
              class="bg-[#C8FFC6] h-8 w-8 rounded-full mr-2 flex justify-center items-center shrink-0"
            >
              <img
                alt=""
                class=""
                src="/src/assets/Icons/travel-insurance.svg"
              />
            </div>
            <div>
              <h3
                class="not-italic text-sm font-medium inline-block mb-1 text-[#101323]"
              >
                Travel Insurance
              </h3>
              <p
                class="text-[#667085] text-xs"
              >
                Travel insurance protects you against unexpected events while traveling, like trip cancellations, lost luggage etc.
              </p>
            </div>
          </div>
          <div
            class="flex justify-between"
          >
            <button
              class="w-[48%] text-white text-sm font-medium bg-[#25D366] rounded-full py-[6px]"
            >
              Sell Policy
            </button>
            <button
              class="w-[48%] text-black text-sm font-medium bg-[#F9FAFB] rounded-full py-[6px]"
            >
              View Policy
            </button>
          </div>
        </div>
        <div
          class="bg-white rounded-2xl p-4 cursor-pointer"
        >
          <div
            class="flex mb-2"
          >
            <div
              class="bg-[#C8FFC6] h-8 w-8 rounded-full mr-2 flex justify-center items-center shrink-0"
            >
              <img
                alt=""
                class=""
                src="/src/assets/Icons/device-insurance.svg"
              />
            </div>
            <div>
              <h3
                class="not-italic text-sm font-medium inline-block mb-1 text-[#101323]"
              >
                Device Insurance
              </h3>
              <p
                class="text-[#667085] text-xs"
              >
                Device insurance covers your device against unexpected events that lead to damage of parts of the phone
              </p>
            </div>
          </div>
          <div
            class="flex justify-between"
          >
            <button
              class="w-[48%] text-white text-sm font-medium bg-[#25D366] rounded-full py-[6px]"
            >
              Sell Policy
            </button>
            <button
              class="w-[48%] text-black text-sm font-medium bg-[#F9FAFB] rounded-full py-[6px]"
            >
              View Policy
            </button>
          </div>
        </div>
      </main>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
